apps/cowswap-frontend/src/modules/swap/services/swapFlow/index.ts

```diff
@@ -71,19 +64,8 @@ export async function swapFlow(
       generatePermitHook,
     })
 
-    if (appDataContainsPermitSigner(orderParams.appData.fullAppData)) {
+    if (callDataContainsPermitSigner(orderParams.appData.fullAppData)) {
       reportPermitWithDefaultSigner(orderParams)
-    } else if (
-      // TODO: remove once we figure out what's adding this to appData in the first place
-      // Last resort in case of a race condition
-      // It should not have a permit in the first place if it's selling native
-      // But there are several cases where it has
-      getIsNativeToken(inputCurrency) &&
```

Comment:
@alfetopito I've removed this condition because we have not only permit hook. I hope it's ok